src/service/fixed-layer.js

```diff
@@ -450,15 +451,38 @@ export class FixedLayer {
           // We shouldn't have too many of `fixed` elements.
           break;
         }
-        this.setupElement_(elements[j], fixedSelector, 'fixed');
+        const el = elements[j];
+        this.removeStyleAttributeIfNecessary_(el);
+        this.setupElement_(el, fixedSelector, 'fixed');
       }
     }
     for (let i = 0; i < stickySelectors.length; i++) {
       const stickySelector = stickySelectors[i];
       const elements = this.ampdoc.getRootNode().querySelectorAll(
           stickySelector);
       for (let j = 0; j < elements.length; j++) {
-        this.setupElement_(elements[j], stickySelector, 'sticky');
+        const element = elements[j];
+        this.removeStyleAttributeIfNecessary_(element);
+        this.setupElement_(element, stickySelector, 'sticky');
+      }
+    }
+  }
+
+  /**
+   * If the given element has a `style` attribute with a top/bottom CSS rule,
+   * remove it and throw a user error. FixedLayer's implementation currently
+   * assumes that those rules cannot be set by the publisher.
+   * @param {!Element} element
+   * @private
+   */
+  removeStyleAttributeIfNecessary_(element) {
+    if (isExperimentOn(this.ampdoc.win, 'inline-styles')) {
+      if (element.hasAttribute('style')
+          && (/top\s*:|bottom\s*:/i).test(element.getAttribute('style'))) {
```

Comment:
@jridgewell Does the fixed-layer algorithm assume that no inline styles exist for CSS rules other than `top` and `bottom`?